### PR TITLE
Feature/inv 172/validation

### PR DESCRIPTION
**Pull request by tikhanovichA**
https://oat-sa.atlassian.net/browse/INV-172

How to test:
1. Open the XML Editor
2. Change some tag name, for example, `assessmentSection` -> `assessmentSectionBAD`

### CODE REVIEW

File: models/classes/validators/XmlSchemaValidator.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+use tao_helpers_form_Validator;
+
+class XmlSchemaValidator extends tao_helpers_form_Validator
+{
+
+    /**
+     * @param string $value
+     * @return bool
+     */
+    public function evaluate($value)
```
Comment (uncleempty) [inline models/classes/validators/XmlSchemaValidator.php:35]: please add param type hint and return type hint

File: models/classes/validators/XmlSchemaValidator.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            $message = '';
+            /** @var \LibXMLError $error */
+            foreach ($errors->getArrayCopy() as $error){
+                $message .= $error->message;
+            }
+            $this->setMessage($message);
+            return false;
+        }
+        return true;
+
```
Comment (uncleempty) [inline models/classes/validators/XmlSchemaValidator.php:51]: please remove excessive blank line

File: models/classes/validators/XmlSchemaValidator.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    /**
+     * Default error message
+     *
+     * @return string
+     */
+    protected function getDefaultMessage()
```
Comment (uncleempty) [inline models/classes/validators/XmlSchemaValidator.php:59]: please add return type hint

File: models/classes/validators/XmlSchemaValidator.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        return __('XML is not valid');
+    }
+
```
Comment (uncleempty) [inline models/classes/validators/XmlSchemaValidator.php:63]: please remove excessive blank line

File: models/classes/forms/XmlEditForm.php [hunk 5]
```diff
@@ -52,7 +53,7 @@ protected function initForm() : void
             'element' => new tao_helpers_form_xhtml_TagWrapper(['tag' => 'div']),
             'error'   => new tao_helpers_form_xhtml_TagWrapper([
                 'tag' => 'div',
-                'cssClass' => 'form-error ui-state-error ui-corner-all hidden'
+                'cssClass' => ' form-error'
```
Comment (uncleempty) [inline models/classes/forms/XmlEditForm.php:56]: should it have a space before css class name?

File: models/classes/validators/XmlSchemaValidator.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+use qtism\data\storage\xml\XmlStorageException;
+use tao_helpers_form_Validator;
+
+class XmlSchemaValidator extends tao_helpers_form_Validator
+{
+
+    /**
+     * @param string $value
+     * @return bool
+     */
```
Comment (uncleempty) [inline models/classes/validators/XmlSchemaValidator.php:34]: also, when you will add type hints - method signature would become self-describing, so no docblock needed :)

**Comment by tikhanovichA** [comment]
> 👍
> Just a few really small changes left

Fixed. Thanks!

**Pull request merged**